src/lib/block_producer/block_producer.ml

```diff
@@ -755,10 +836,11 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
       in
       let production_supervisor = Singleton_supervisor.create ~task:produce in
       let scheduler = Singleton_scheduler.create time_controller in
-      let rec check_next_block_timing () =
+      let vrf_evaluation_state = Vrf_evaluation_state.create () in
+      let rec check_next_block_timing slot epoch () =
         trace_recurring "check next block timing" (fun () ->
             (* See if we want to change keypairs *)
-            let keypairs =
+            let _keypairs =
```

Comment:
I think there's more work required because of the new polling logic. Slots that have already been picked up by the daemon won’t be reevaluated by Vrf_evaluator. Any update to the keys will only reflect in slots that are yet to be evaluated (or the ones that haven’t been picked by the daemon). So, if the keys were swapped then old slots are not valid. If new keys were added there could be some new slots won between the winning slots from the existing keys that are already polled.  So there is some state management required on the block producer side as well.